### PR TITLE
Fix exception that ocurrs when outputting failure message for non-heterogenous data

### PR DESCRIPTION
**Pull request by phillipm**
Fix for https://github.com/marick/Midje/issues/265

`emission.plugins.util/nested-sort` tries to be smart about what it sorts and what it doesn't sort by partitioning it by class. The issue is that `(class [1])` is the same as `(class [:a])` but you can't sort `[[1] [:a]]`.

Since this is error printing code that simply attempts to prettify the output, and implementing a correct sorter for this corner case would be a bunch of work, I've opted to default to showing unsorted output if sorting fails.

### CODE REVIEW

File: test/midje/emission/plugins/t_util.clj [hunk 1]
```diff
@@ -15,6 +15,8 @@
   ;; Note ordering
   (attractively-stringified-value {:b 2 :a 1}) => "{:a 1, :b 2}"
   (attractively-stringified-value #{9 6 2 7 1 3}) => "#{1 2 3 6 7 9}"
+  (read-string (attractively-stringified-value #{[1] [:a]})) => #{[1] [:a]}
+  (read-string (attractively-stringified-value {[1] "1" [:a] "a"})) => {[1] "1" [:a] "a"}
```
Comment (phillipm) [inline test/midje/emission/plugins/t_util.clj]: A tricky way to test this data-to-string function without relying on the ordering of the data.
I wasn't able to create a test at the `fact` level because I wasn't able to get `silent-fact`/`note-that` to work in my favor in this context..

Comment (rafaeldff) [inline test/midje/emission/plugins/t_util.clj]: Would `=> (some-checker "#{[1] [:a]}" "#{[:a] [1]}")` work for you in this case?
